pkg/infrastructure/infinispan.go

```diff
@@ -49,3 +53,49 @@ func IsInfinispanOperatorAvailable(cli *client.Client, namespace string) (bool,
 	log.Debugf("Looks like Infinispan Operator is not available in the namespace %s", namespace)
 	return false, nil
 }
+
+// GetInfinispanServiceURI fetches for the Infinispan service linked with the given KogitoInfra and returns a formatted URI
+func GetInfinispanServiceURI(cli *client.Client, infra *v1alpha1.KogitoInfra) (uri string, err error) {
+	if &infra == nil || &infra.Status == nil || &infra.Status.Infinispan == nil {
+		return "", nil
+	}
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: infra.Status.Infinispan.Service, Namespace: infra.Namespace},
+	}
+	exists := false
+	if exists, err = kubernetes.ResourceC(cli).Fetch(service); err != nil {
+		return "", err
+	}
+
+	if exists && len(service.Spec.Ports) > 0 {
+		return fmt.Sprintf("%s:%d", service.Name, service.Spec.Ports[0].TargetPort.IntVal), nil
```

Comment:
Here you are getting the first port on the service.
On my test: 
`$ oc get svc kogito-infinispan -o yaml`
gives:
```
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2019-11-27T10:19:39Z"
  name: kogito-infinispan
  namespace: kogito-165
  ownerReferences:
  - apiVersion: infinispan.org/v1
    blockOwnerDeletion: true
    controller: true
    kind: Infinispan
    name: kogito-infinispan
    uid: 6b2c9a14-10ff-11ea-821f-52fdfc072182
  resourceVersion: "246493"
  selfLink: /api/v1/namespaces/kogito-165/services/kogito-infinispan
  uid: 6b47880a-10ff-11ea-821f-52fdfc072182
spec:
  clusterIP: 172.30.33.139
  ports:
  - name: http
    port: 8080
    protocol: TCP
    targetPort: 8080
  - name: hotrod
    port: 11222
    protocol: TCP
    targetPort: 11222
  selector:
    app: infinispan-pod
    clusterName: kogito-infinispan
    infinispan_cr: kogito-infinispan
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```
And so is my pod launch with environment:
`QUARKUS_INFINISPAN_CLIENT_SERVER_LIST=kogito-infinispan:8080`

Shouldn't it be `kogito-infinispan:11222` ?

Reply:
That’s nice. I’ll get the default port instead. The problem is that the version of the operator I used haven’t set more than one port. Are you using the Data Grid operator?

Reply:
Data Grid Operator ... Maybe that is the problem...

Reply:
Here's the version:
https://github.com/infinispan/infinispan-operator/blob/0.3.2/pkg/controller/infinispan/infinispan_controller.go#L418

Totally wrong :p

They changed in the 1.0.1, but I'll make the code compatible with both..... because we can.